server/src/mds3.load.js

```diff
@@ -479,77 +479,41 @@ async function geneExpressionClustering(data, q, ds) {
 
 	const Rinputfile = path.join(serverconfig.cachedir, Math.random().toString() + '.json')
 	await write_file(Rinputfile, JSON.stringify(inputData))
-	const Routput = await lines2R(path.join(serverconfig.binpath, 'utils/fastclust.R'), [], [Rinputfile])
+	const Routput = JSON.parse(await lines2R(path.join(serverconfig.binpath, 'utils/hclust.R'), [], [Rinputfile]))
 	fs.unlink(Rinputfile, () => {})
-	//const r_output_lines = Routput.trim().split('\n')
-	//console.log('r_output_lines:', r_output_lines)
 
 	let row_coordinates = []
+	for (const item of JSON.parse(Routput['RowNodeJson'])) {
```

Comment:
should only JSON.parse() once at line 482. e.g. to get rid of this json.parse, delete `toJSON()` in line 105 hclust.r